provider/provider.go

```diff
@@ -127,10 +232,40 @@ func For(name string, config dict.Dicter) (Tiler, error) {
 		err.Name = name
 		return nil, err
 	}
+	// need to check to see if p if of type mvt provider
+	if p.init == nil {
+		return nil, ErrInvalidProviderType{
+			Name:           name,
+			Type:           TypeStd,
+			KnownProviders: Drivers(ProviderFilterStd),
+		}
+	}
 
 	return p.init(config)
 }
 
+// MVTFor function returns a configured mvt provider of the given type, provided the correct config map.
+func MVTFor(name string, config dict.Dicter) (MVTTiler, error) {
+	if providers == nil {
+		return nil, ErrUnknownProvider{}
```

Comment:
Should this error be expanded to `ErrUnknownProvider{KnownProviders: Drivers(ProviderFilterMvt)}`? If not, maybe review the `STDFor` function so these two scenarios align.

Reply:
I think I can get rid of the STDFor and MVTFor, but you are correct it should be filtered.